crates/valence/examples/time_travel.rs

```diff
@@ -0,0 +1,108 @@
+use valence::client::despawn_disconnected_clients;
+use valence::client::hand_swing::HandSwingEvent;
+use valence::client::message::SendMessage;
+use valence::inventory::HeldItem;
+use valence::prelude::*;
+use valence::world_time::{ChangeTrackingTimeBroadcast, LinearTimeTicking, WorldTime};
+
+const SPAWN_Y: i32 = 64;
+
+pub fn main() {
+    tracing_subscriber::fmt().init();
+
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .add_system(init_clients)
+        .add_system(despawn_disconnected_clients)
+        .add_system(handle_adjust_speed)
+        .add_system(handle_display_time)
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    server: Res<Server>,
+    biomes: Res<BiomeRegistry>,
+    dimensions: Res<DimensionTypeRegistry>,
+) {
+    let mut instance = Instance::new(ident!("overworld"), &dimensions, &biomes, &server);
+
+    for z in -5..5 {
+        for x in -5..5 {
+            instance.insert_chunk([x, z], Chunk::default());
+        }
+    }
+
+    for z in -25..25 {
+        for x in -25..25 {
+            instance.set_block([x, SPAWN_Y, z], BlockState::GRASS_BLOCK);
+        }
+    }
+
+    commands.spawn((
+        instance,
+        WorldTime {
+            time_of_day: -1,
```

Comment:
Vanilla server will set time_of_day to `-1` if you use `/gamerule doDaylightCycle false` and `/time set 0`

Reply:
Maybe you could make that more explicit here by using `set_client_time_ticking`. But it looks like `set_client_time_ticking` is wrong when the time of day is zero because `-0.abs()` is still `0`.

Reply:
Yeah, currently there is no way around that I think. Mojang devs probably knew it, so they use -1 as a substitute